### PR TITLE
[REF] mail: adapts the attachment component for extension

### PR DESCRIPTION
**Pull request by ThanhDodeurOdoo**
This commit adds two changes to the Attachment component:

* increase the specificity of an overflow rule so it is not applied
when unnecessary.

* replaces the usage of the native `event.stopPropagation()`
by the event handling of mail.utils.

part of task-2360547

### CODE REVIEW

**Comment by ThanhDodeurOdoo** [comment]
related to https://github.com/odoo/enterprise/pull/15598

**Review by alexkuhn — CHANGES_REQUESTED** [review]

File: addons/mail/static/src/components/attachment/attachment.scss
Comment (alexkuhn) [inline addons/mail/static/src/components/attachment/attachment.scss:26]: `o_isInComposer` => `o-isInComposer`

File: addons/mail/static/src/components/attachment/attachment.scss
Comment (alexkuhn) [inline addons/mail/static/src/components/attachment/attachment.scss:28]: empty newline after

**Review by alexkuhn — CHANGES_REQUESTED** [review]

**Review by alexkuhn — COMMENTED** [review]
LGTM, waiting for enterprise PR being good too.

File: addons/mail/static/src/components/attachment/attachment.js
Comment (alexkuhn) [inline addons/mail/static/src/components/attachment/attachment.js:138]: Would clarify exactly action that is being handled here: `Attachment.clickUnlink`.

File: addons/mail/static/src/components/attachment/attachment.js
Comment (alexkuhn) [inline addons/mail/static/src/components/attachment/attachment.js:111]: Same: `Attachment.clickDownload`.

File: addons/mail/static/src/components/attachment/attachment.js
Comment (alexkuhn) [inline addons/mail/static/src/components/attachment/attachment.js:122]: I'm not fond of || in conditional for preemptive return. It makes code harder to read.

I prefer this:

```js
if (!this.attachment.isViewable) {
    return;
}
if (isEventHandled(ev, 'Attachment.clickDownload')) {
    return;
}
if (isEventHandled(ev, 'Attachment.clickUnlink')) {
    return;
}
```

**Review by alexkuhn — COMMENTED** [review]
LGTM

**Comment by alexkuhn** [comment]
_(waiting for enterprise requested changes + approval)_

File: addons/mail/static/tests/chatter_tests.js
Comment (alexkuhn) [inline addons/mail/static/tests/chatter_tests.js:118]: You need to fix this in other tests of this file. Enterprise branch is red because new fetches affect these tests.

File: addons/mail/static/tests/chatter_tests.js
Comment (alexkuhn) [inline addons/mail/static/tests/chatter_tests.js:238]: Would prefer `switch/case` in `mockRPC` like with the enterprise branch.
Also could turn all of them as route rather than having mix of routes and methods.

I don't mind listing the RPCs twice. I prefer that introducing this variable.

**Review by alexkuhn — CHANGES_REQUESTED** [review]

File: addons/mail/static/src/components/attachment/attachment.js
Comment (seb-odoo) [inline addons/mail/static/src/components/attachment/attachment.js:123]: `²`?

File: addons/mail/static/src/components/attachment/attachment.scss
Comment (seb-odoo) [inline addons/mail/static/src/components/attachment/attachment.scss:28]: What's the goal here? And why would we allow it to overflow in any other case?

File: addons/mail/static/tests/chatter_tests.js
Comment (seb-odoo) [inline addons/mail/static/tests/chatter_tests.js:116]: Don't ever use a regex if you can use anything else (such as in this case ideally a strict equality or using `includes`). Same in whole diff.
Regex is usually much slower, and always hard to understand and maintain.

Comment (seb-odoo) [reply addons/mail/static/tests/chatter_tests.js:116]: Btw what are the extra queries that you are trying to filter out?
I like that this test showed us exactly what was happening, even though its name is completely wrong for this purpose.

**Review by seb-odoo — CHANGES_REQUESTED** [review]
Quick check of community part

File: addons/mail/static/src/components/attachment/attachment.scss
Comment (seb-odoo) [reply addons/mail/static/src/components/attachment/attachment.scss:28]: Actually I should reformulate the question, why is there an overflow hidden in the first place?

`overflow: hidden` should usually be avoided all-together.
The only legitimate use case off the top of by head is to be able to add ellipsis to text when we don't want it to wrap to new line.
The better alternative is to ensure elements that can get too big don't get too big in the first place, with a proper max-width, flex shrink, wrapping, or whatever else is necessary.

File: addons/mail/static/tests/chatter_tests.js
Comment (seb-odoo) [reply addons/mail/static/tests/chatter_tests.js:116]: Ok I saw the extra rpc in enterprise, and my first feeling is that we should not have them (see enterprise PR comment) and this whole diff should be reverted.

File: addons/mail/static/src/components/attachment/attachment.scss
Comment (ThanhDodeurOdoo) [reply addons/mail/static/src/components/attachment/attachment.scss:28]: @seb-odoo I think that the purpose was to hide the "big" cross button to delete attachments in the composer, it is hidden out of the card and it seems that it's a trick to make it slide in the card when we hover it.

File: addons/mail/static/src/components/attachment/attachment.scss
Comment (seb-odoo) [inline addons/mail/static/src/components/attachment/attachment.scss:45]: `.o-pretty` is top level style? That seems strange and/or an unclear class name.

File: addons/mail/static/src/components/attachment/attachment.scss
Comment (seb-odoo) [inline addons/mail/static/src/components/attachment/attachment.scss:49]: map-get($sizes, 100);

File: addons/mail/static/src/components/attachment/attachment.scss
Comment (seb-odoo) [inline addons/mail/static/src/components/attachment/attachment.scss:60]: map-get($sizes, 100);

File: addons/mail/static/src/components/attachment/attachment.scss
Comment (seb-odoo) [inline addons/mail/static/src/components/attachment/attachment.scss:173]: $white

File: addons/mail/static/src/components/attachment/attachment.xml
Comment (seb-odoo) [inline addons/mail/static/src/components/attachment/attachment.xml:88]: Why do you need a div into a div as opposed to previous code which only had one?

**Review by seb-odoo — COMMENTED** [review]
Quick check (I'll need to check the scss file in more details later)

Also note there is currently a conflict

**Comment by C3POdoo** [comment]
Dear @ThanhDodeurOdoo,

Thank you for your contribution but we are closing it due to inactivity.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**
